core/user-flow.js

```diff
@@ -221,16 +195,41 @@ class UserFlow {
   }
 
   /**
-   * @return {LH.UserFlow.FlowArtifacts}
+   * @return {Promise<LH.UserFlow.FlowArtifacts>}
    */
-  createArtifactsJson() {
+  async createArtifactsJson() {
     return {
       gatherSteps: this._gatherSteps,
-      name: this.name,
+      name: this._options?.name,
     };
   }
 }
 
+/**
+ * @param {string} longUrl
+ * @returns {string}
+ */
+function shortenUrl(longUrl) {
+  const url = new URL(longUrl);
+  return `${url.hostname}${url.pathname}`;
+}
+
+/**
+ * @param {LH.Artifacts} artifacts
+ * @return {string}
```

Comment:
It's surprising to me that TS agrees this is `string` instead of `string=`. I know the switch covers all enumerations of gather mode, but ... javascript 🙃 

wdyt of 1) throwing an error in a default case or 2) removing the switch and capitalizing the gather mode to get the desired string?

Reply:
I prefer option 1. Option 2 doesn't cover i18n and I don't want to support custom gather modes.

Reply:
Wait, how does the current code cover i18n?

Reply:
> Wait, how does the current code cover i18n? should this function be a report UIString? (if so, this should be a followup)

It doesn't support i18n, we should do it in a followup.